### PR TITLE
Added some tests, update phpunit.xml.dist to include whitelist for co…

### PR DESCRIPTION
**Pull request by SamMousa**
Improved test coverage, related to #261

### CODE REVIEW

**Comment by joelwurtz** [comment]
Looks good, Can you fix cs so i can merge this ?

**Comment by SamMousa** [comment]
This is ready :)

**Comment by joelwurtz** [comment]
Conflict sorry xD

**Comment by SamMousa** [comment]
GitHub appears a bit buggy; after merging 2.0-dev back into this it still told me I had conflicts 8-)

Close and reopen seems to have fix it..

File: tests/Context/ContextTest.php [hunk 1]
```diff
@@ -52,5 +52,15 @@ public function testTarFailed(): void
         } finally {
             \putenv("PATH=$path");
         }
+
```
Comment (joelwurtz) [inline tests/Context/ContextTest.php]: Missing bracket close :/

Comment (SamMousa) [reply tests/Context/ContextTest.php]: I don't see it?? 
````
         } finally {
             \putenv("PATH=$path");
         }
````

**Comment by SamMousa** [comment]
Finally, github happy, travis happpy :)

**Comment by joelwurtz** [comment]
Thanks a lot

**Pull request merged**